ca_tools/scripts/control_node.py

```diff
@@ -0,0 +1,173 @@
+        euler = euler_from_quaternion(q_arr)
+        self._my_pose.theta = euler[2]
+        self._move()
+
+    def _move(self):
+        if self._state == TURNING:
+            self._turning()
+        elif self._state == STOP_TURNING:
+            self._stop_turning()
+        elif self._state == FORWARD:
+            self._forward()
+        elif self._state == STOP_FORWARD:
+            self._stop_forward()
+
+    def _turning(self):
+
+        if (self._reached_angle()):
+            self._stop()
+            self._state = STOP_TURNING
+
+        else:
+            aux = Twist()
+            if (abs(self._my_anglegoal-self._my_pose.theta) > pi):
+                negateflag = -1.0
+            else:
+                negateflag = 1.0
+
+            if (self._my_anglegoal-self._my_pose.theta) < 0:
+                aux.angular.z = negateflag*-ANGULAR_VEL
+            else:
+                aux.angular.z = ANGULAR_VEL
+
+            rospy.loginfo("negateflag is %f", negateflag)
+            self._my_pub.publish(aux)
+
+    def _stop_turning(self):
+        self._stop()
+        self._move_forward()
+        self._state = FORWARD
+
+    def _stop(self):.
+        aux = Twist()
+        self._my_pub.publish(aux)
+
+    def _move_forward(self):
+        aux = Twist()
+        aux.linear.x = LINEAR_VEL
+        self._my_pub.publish(aux)
+
+    def _forward(self):
+        if (self._reached_position()):
+            self._stop()
+            self._state = STOP_FORWARD
+            self._goal_num += 1
+            self._goal_num %= 4
+        else:
+            self._set_goal_angle()
+            if (not self._reached_angle()):
+                self._state = STOP_FORWARD
+                self._stop()
+            else:
+                self._move_forward()
+
+    def _diff_angle(self):
+        return (self._my_anglegoal-self._my_pose.theta)
+
+    def _angle_is_big(self):
+        return ((self._my_anglegoal-self._my_pose.theta) > BIG_ANGLE_THRESHOLD) and ((self._my_anglegoal-self._my_pose.theta) < ((2*pi)-BIG_ANGLE_THRESHOLD))
+
+    def _stop_forward(self):
+        self._stop()
+        self._set_goal_angle()
+        self._state = TURNING
+
+    def _set_goal_angle(self):
+        self._my_anglegoal = math.atan2(
+            (self._my_goals[self._goal_num].y-self._my_pose.y), self._my_goals[self._goal_num].x-self._my_pose.x)
+        rospy.loginfo("Difference between angles is %f,",
+                      (self._my_anglegoal-self._my_pose.theta))
+
+    def _reached_angle(self):
+        return ((abs(self._my_anglegoal-self._my_pose.theta)) < ANGLE_TOLERANCE or (abs(self._my_anglegoal-self._my_pose.theta)) > (2*pi-ANGLE_TOLERANCE))
+
+    def _reached_position(self):
+        reached_y = (
+            (abs(self._my_goals[self._goal_num].y-self._my_pose.y)) < DISTANCE_TOLERANCE)
+        reached_x = (
+            abs(self._my_goals[self._goal_num].x-self._my_pose.x) < DISTANCE_TOLERANCE)
+        return (reached_x and reached_y)
```

Comment:
This is not an intuitive notion of distance. Why not use the euclidean distance?